### PR TITLE
Make number of repetitions configurable

### PR DESCRIPTION
**Pull request by chrzaszcz**
Additional fixes:
- fix deps to work with R18 and accept later versions of escalus
- clean relx setup

### CODE REVIEW

File: src/amoc_user.erl [hunk 1]
```diff
@@ -9,6 +9,10 @@
 -define(REPEAT_INTERVAL,
     application:get_env(amoc, repeat_interval, ?REPEAT_INTERVAL_DEFAULT)).
 
+-define(REPEAT_NUM_DEFAULT, infinity).
+-define(REPEAT_NUM,
+    application:get_env(amoc, repeat_num, ?REPEAT_NUM_DEFAULT)).
```
Comment (studzien) [inline src/amoc_user.erl]: Wouldn't a function definition (e.g. ``repeat_num/0``) be better for this purpose?
Yes, the repeat interval value is defined with macros as well, but I believe it also should be changed.

Comment (studzien) [inline src/amoc_user.erl]: See: https://medium.com/@erszcz/when-not-to-use-macros-in-erlang-1d3f10d377f

Comment (chrzaszcz) [inline src/amoc_user.erl]: I will kill this monstrosity with fire.

Comment (studzien) [inline src/amoc_user.erl]: 😎

**Pull request merged**